test/val/val_id_test.cpp

```diff
@@ -3550,7 +3550,7 @@ TEST_F(ValidateIdWithMessage, OpFunctionResultTypeBad) {
   EXPECT_EQ(SPV_ERROR_INVALID_ID, ValidateInstructions());
   EXPECT_THAT(getDiagnosticString(),
               HasSubstr("OpFunction Result Type <id> '2' does not match the "
-                        "Function Type <id> '2's return type."));
+                        "Function Type's return type <id> '1'."));
```

Comment:
The error seemed bad before as it referred to id 2 twice. I updated it to describe the ids of the actual mismatch.